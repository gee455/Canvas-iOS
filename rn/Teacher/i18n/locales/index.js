//
// Copyright (C) 2016-present Instructure, Inc.
//
// This program is free software: you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation, version 3 of the License.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see <http://www.gnu.org/licenses/>.
//

/* @flow */

export default ({
  ar: require('./ar.json'),
  da: require('./da.json'),
  de: require('./de.json'),
  'en-AU': require('./en_AU.json'),
  'en-GB': require('./en_GB.json'),
  en: require('./en.json'),
  es: require('./es.json'),
  'fr-CA': require('./fr_CA.json'),
  fr: require('./fr.json'),
  ht: require('./ht.json'),
  ja: require('./ja.json'),
  mi: require('./mi.json'),
  nb: require('./nb.json'),
  nl: require('./nl.json'),
  pl: require('./pl.json'),
  'pt-BR': require('./pt_BR.json'),
  pt: require('./pt.json'),
  ru: require('./ru.json'),
  sv: require('./sv.json'),
  'zh-HK': require('./zh_HK.json'),
  zh: require('./zh.json'),
}: { [string]: { [string]: { message: string, description?: string } } })
