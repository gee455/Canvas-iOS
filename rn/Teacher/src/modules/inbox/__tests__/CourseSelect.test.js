//
// Copyright (C) 2016-present Instructure, Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

// @flow

import React from 'react'
import {
  CourseSelect,
  mapStateToProps,
  shouldRefresh,
  doRefresh,
  isRefreshing } from '../CourseSelect'
import renderer from 'react-test-renderer'
import explore from '../../../../test/helpers/explore'
import i18n from 'format-message'

jest.mock('TouchableHighlight', () => 'TouchableHighlight')

let template = {
  ...require('../../../__templates__/course'),
  ...require('../../../redux/__templates__/app-state'),
  ...require('../../../__templates__/helm'),
}

let c1 = template.course({
  is_favorite: true,
  id: '1',
})

let c2 = template.course({
  is_favorite: false,
  id: '2',
})

let c3 = template.course({
  is_favorite: false,
  id: '3',
})

let defaultProps = {
  navigator: template.navigator({
    dismiss: jest.fn(),
  }),
  onSelect: jest.fn(),
  pending: false,
  sections: [
    {
      key: 0,
      title: 'favorites',
      data: [c1],
    },
    {
      key: 1,
      title: 'courses',
      data: [c2, c3],
    },
  ],
}

describe('CourseSelect', () => {
  beforeEach(() => jest.resetAllMocks())

  it('renders', () => {
    let tree = renderer.create(
      <CourseSelect {...defaultProps} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders and then selects a course and then dismisses', () => {
    const component = renderer.create(
      <CourseSelect {...defaultProps} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    const courseRow = explore(tree).selectByID(`inbox.course-select.course-${c1.id}`) || {}
    courseRow.props.onPress()
    expect(defaultProps.onSelect).toHaveBeenCalled()
  })

  it('mapStateToProps', () => {
    const appState = template.appState({
      entities: {
        courses: {
          [c1.id]: {
            course: c1,
          },
          [c2.id]: {
            course: c2,
          },
          [c3.id]: {
            course: c3,
          },
        },
      },
    })

    let result = mapStateToProps(appState)
    expect(result).toMatchObject({
      sections: [
        {
          key: 0,
          title: i18n('Favorite Courses'),
          data: [c1],
        },
        {
          key: 1,
          title: i18n('Courses'),
          data: [c2, c3],
        },
      ],
    })
  })

  it('refresh functions', () => {
    const props = {
      refreshCourses: jest.fn(),
    }

    expect(shouldRefresh()).toEqual(true)
    doRefresh(props)
    expect(props.refreshCourses).toHaveBeenCalled()
    expect(isRefreshing({ pending: true })).toEqual(true)
  })
})
