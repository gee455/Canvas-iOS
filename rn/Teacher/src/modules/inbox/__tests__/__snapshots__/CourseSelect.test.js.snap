// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseSelect renders 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    ItemSeparatorComponent={undefined}
    data={
      Array [
        Object {
          "data": Array [
            Object {
              "account_id": "1",
              "course_code": "rn 101",
              "default_view": "wiki",
              "end_at": "2030-02-08T23:13:39Z",
              "enrollments": Array [
                Object {
                  "enrollment_state": "active",
                  "role": "TeacherEnrollment",
                  "role_id": "1",
                  "type": "teacher",
                  "user_id": "1",
                },
              ],
              "id": "1",
              "image_download_url": "https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg",
              "is_favorite": true,
              "name": "Learn React Native",
              "permissions": Object {
                "viewAllGrades": true,
                "viewAnalytics": true,
              },
              "short_name": "rn",
              "term": Object {
                "created_at": "2018-02-08T23:13:39Z",
                "end_at": "2030-02-08T23:13:39Z",
                "id": "1",
                "name": "Default Term",
                "start_at": "2018-02-08T23:13:39Z",
                "workflow_state": "available",
              },
              "workflow_state": "available",
            },
          ],
          "key": 0,
          "title": "favorites",
        },
        Object {
          "data": Array [
            Object {
              "account_id": "1",
              "course_code": "rn 101",
              "default_view": "wiki",
              "end_at": "2030-02-08T23:13:39Z",
              "enrollments": Array [
                Object {
                  "enrollment_state": "active",
                  "role": "TeacherEnrollment",
                  "role_id": "1",
                  "type": "teacher",
                  "user_id": "1",
                },
              ],
              "id": "2",
              "image_download_url": "https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg",
              "is_favorite": false,
              "name": "Learn React Native",
              "permissions": Object {
                "viewAllGrades": true,
                "viewAnalytics": true,
              },
              "short_name": "rn",
              "term": Object {
                "created_at": "2018-02-08T23:13:39Z",
                "end_at": "2030-02-08T23:13:39Z",
                "id": "1",
                "name": "Default Term",
                "start_at": "2018-02-08T23:13:39Z",
                "workflow_state": "available",
              },
              "workflow_state": "available",
            },
            Object {
              "account_id": "1",
              "course_code": "rn 101",
              "default_view": "wiki",
              "end_at": "2030-02-08T23:13:39Z",
              "enrollments": Array [
                Object {
                  "enrollment_state": "active",
                  "role": "TeacherEnrollment",
                  "role_id": "1",
                  "type": "teacher",
                  "user_id": "1",
                },
              ],
              "id": "3",
              "image_download_url": "https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg",
              "is_favorite": false,
              "name": "Learn React Native",
              "permissions": Object {
                "viewAllGrades": true,
                "viewAnalytics": true,
              },
              "short_name": "rn",
              "term": Object {
                "created_at": "2018-02-08T23:13:39Z",
                "end_at": "2030-02-08T23:13:39Z",
                "id": "1",
                "name": "Default Term",
                "start_at": "2018-02-08T23:13:39Z",
                "workflow_state": "available",
              },
              "workflow_state": "available",
            },
          ],
          "key": 1,
          "title": "courses",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onViewableItemsChanged={undefined}
    renderItem={[Function]}
    renderSectionHeader={[Function]}
    scrollEventThrottle={50}
    sections={
      Array [
        Object {
          "data": Array [
            Object {
              "account_id": "1",
              "course_code": "rn 101",
              "default_view": "wiki",
              "end_at": "2030-02-08T23:13:39Z",
              "enrollments": Array [
                Object {
                  "enrollment_state": "active",
                  "role": "TeacherEnrollment",
                  "role_id": "1",
                  "type": "teacher",
                  "user_id": "1",
                },
              ],
              "id": "1",
              "image_download_url": "https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg",
              "is_favorite": true,
              "name": "Learn React Native",
              "permissions": Object {
                "viewAllGrades": true,
                "viewAnalytics": true,
              },
              "short_name": "rn",
              "term": Object {
                "created_at": "2018-02-08T23:13:39Z",
                "end_at": "2030-02-08T23:13:39Z",
                "id": "1",
                "name": "Default Term",
                "start_at": "2018-02-08T23:13:39Z",
                "workflow_state": "available",
              },
              "workflow_state": "available",
            },
          ],
          "key": 0,
          "title": "favorites",
        },
        Object {
          "data": Array [
            Object {
              "account_id": "1",
              "course_code": "rn 101",
              "default_view": "wiki",
              "end_at": "2030-02-08T23:13:39Z",
              "enrollments": Array [
                Object {
                  "enrollment_state": "active",
                  "role": "TeacherEnrollment",
                  "role_id": "1",
                  "type": "teacher",
                  "user_id": "1",
                },
              ],
              "id": "2",
              "image_download_url": "https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg",
              "is_favorite": false,
              "name": "Learn React Native",
              "permissions": Object {
                "viewAllGrades": true,
                "viewAnalytics": true,
              },
              "short_name": "rn",
              "term": Object {
                "created_at": "2018-02-08T23:13:39Z",
                "end_at": "2030-02-08T23:13:39Z",
                "id": "1",
                "name": "Default Term",
                "start_at": "2018-02-08T23:13:39Z",
                "workflow_state": "available",
              },
              "workflow_state": "available",
            },
            Object {
              "account_id": "1",
              "course_code": "rn 101",
              "default_view": "wiki",
              "end_at": "2030-02-08T23:13:39Z",
              "enrollments": Array [
                Object {
                  "enrollment_state": "active",
                  "role": "TeacherEnrollment",
                  "role_id": "1",
                  "type": "teacher",
                  "user_id": "1",
                },
              ],
              "id": "3",
              "image_download_url": "https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg",
              "is_favorite": false,
              "name": "Learn React Native",
              "permissions": Object {
                "viewAllGrades": true,
                "viewAnalytics": true,
              },
              "short_name": "rn",
              "term": Object {
                "created_at": "2018-02-08T23:13:39Z",
                "end_at": "2030-02-08T23:13:39Z",
                "id": "1",
                "name": "Default Term",
                "start_at": "2018-02-08T23:13:39Z",
                "workflow_state": "available",
              },
              "workflow_state": "available",
            },
          ],
          "key": 1,
          "title": "courses",
        },
      ]
    }
    stickyHeaderIndices={
      Array [
        0,
        3,
      ]
    }
    stickySectionHeadersEnabled={true}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityTraits="header"
          style={
            Array [
              Object {
                "backgroundColor": "#F5F5F5",
                "flex": 1,
                "height": "auto",
                "justifyContent": "center",
                "paddingLeft": 16,
                "paddingRight": 8,
                "paddingVertical": 4,
              },
              Object {
                "borderBottomColor": "#C7CDD1",
                "borderBottomWidth": 0.5,
              },
            ]
          }
        >
          <Text
            accessibilityTraits="header"
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontFamily": ".SFUIDisplay-semibold",
                  "fontSize": 20,
                },
                Object {
                  "backgroundColor": "#F5F5F5",
                  "color": "#73818C",
                  "fontSize": 14,
                  "fontWeight": "600",
                },
              ]
            }
          >
            favorites
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <TouchableHighlight
          accessibilityLabel={undefined}
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          onPress={[Function]}
          style={
            Array [
              Object {
                "borderTopColor": "#C7CDD1",
                "borderTopWidth": 0.5,
              },
              Object {
                "borderBottomColor": "#C7CDD1",
                "borderBottomWidth": 0.5,
              },
            ]
          }
          testID="inbox.course-select.course-1"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "flex": 1,
                  "flexDirection": "row",
                  "minHeight": 54,
                  "paddingBottom": 12,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 12,
                },
                Object {
                  "backgroundColor": "white",
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                  },
                  Object {
                    "marginLeft": 0,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={0}
                style={
                  Array [
                    Object {
                      "fontFamily": ".SFUIDisplay",
                    },
                    Object {
                      "color": "#2D3B45",
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "fontWeight": "600",
                      },
                    ],
                    Object {
                      "fontFamily": ".SFUIDisplay-semibold",
                    },
                  ]
                }
              >
                Learn React Native
              </Text>
            </View>
          </View>
        </TouchableHighlight>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityTraits="header"
          style={
            Array [
              Object {
                "backgroundColor": "#F5F5F5",
                "flex": 1,
                "height": "auto",
                "justifyContent": "center",
                "paddingLeft": 16,
                "paddingRight": 8,
                "paddingVertical": 4,
              },
              Object {
                "borderBottomColor": "#C7CDD1",
                "borderBottomWidth": 0.5,
              },
            ]
          }
        >
          <Text
            accessibilityTraits="header"
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontFamily": ".SFUIDisplay-semibold",
                  "fontSize": 20,
                },
                Object {
                  "backgroundColor": "#F5F5F5",
                  "color": "#73818C",
                  "fontSize": 14,
                  "fontWeight": "600",
                },
              ]
            }
          >
            courses
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <TouchableHighlight
          accessibilityLabel={undefined}
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          onPress={[Function]}
          style={
            Array [
              Object {
                "borderTopColor": "#C7CDD1",
                "borderTopWidth": 0.5,
              },
              Object {
                "borderBottomColor": "#C7CDD1",
                "borderBottomWidth": 0.5,
              },
            ]
          }
          testID="inbox.course-select.course-2"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "flex": 1,
                  "flexDirection": "row",
                  "minHeight": 54,
                  "paddingBottom": 12,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 12,
                },
                Object {
                  "backgroundColor": "white",
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                  },
                  Object {
                    "marginLeft": 0,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={0}
                style={
                  Array [
                    Object {
                      "fontFamily": ".SFUIDisplay",
                    },
                    Object {
                      "color": "#2D3B45",
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "fontWeight": "600",
                      },
                    ],
                    Object {
                      "fontFamily": ".SFUIDisplay-semibold",
                    },
                  ]
                }
              >
                Learn React Native
              </Text>
            </View>
          </View>
        </TouchableHighlight>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <TouchableHighlight
          accessibilityLabel={undefined}
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          onPress={[Function]}
          style={
            Array [
              undefined,
              Object {
                "borderBottomColor": "#C7CDD1",
                "borderBottomWidth": 0.5,
              },
            ]
          }
          testID="inbox.course-select.course-3"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "flex": 1,
                  "flexDirection": "row",
                  "minHeight": 54,
                  "paddingBottom": 12,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 12,
                },
                Object {
                  "backgroundColor": "white",
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                  },
                  Object {
                    "marginLeft": 0,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={0}
                style={
                  Array [
                    Object {
                      "fontFamily": ".SFUIDisplay",
                    },
                    Object {
                      "color": "#2D3B45",
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "fontWeight": "600",
                      },
                    ],
                    Object {
                      "fontFamily": ".SFUIDisplay-semibold",
                    },
                  ]
                }
              >
                Learn React Native
              </Text>
            </View>
          </View>
        </TouchableHighlight>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;

exports[`CourseSelect renders and then selects a course and then dismisses 1`] = `
<RCTSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTScrollView
    ItemSeparatorComponent={undefined}
    data={
      Array [
        Object {
          "data": Array [
            Object {
              "account_id": "1",
              "course_code": "rn 101",
              "default_view": "wiki",
              "end_at": "2030-02-08T23:13:39Z",
              "enrollments": Array [
                Object {
                  "enrollment_state": "active",
                  "role": "TeacherEnrollment",
                  "role_id": "1",
                  "type": "teacher",
                  "user_id": "1",
                },
              ],
              "id": "1",
              "image_download_url": "https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg",
              "is_favorite": true,
              "name": "Learn React Native",
              "permissions": Object {
                "viewAllGrades": true,
                "viewAnalytics": true,
              },
              "short_name": "rn",
              "term": Object {
                "created_at": "2018-02-08T23:13:39Z",
                "end_at": "2030-02-08T23:13:39Z",
                "id": "1",
                "name": "Default Term",
                "start_at": "2018-02-08T23:13:39Z",
                "workflow_state": "available",
              },
              "workflow_state": "available",
            },
          ],
          "key": 0,
          "title": "favorites",
        },
        Object {
          "data": Array [
            Object {
              "account_id": "1",
              "course_code": "rn 101",
              "default_view": "wiki",
              "end_at": "2030-02-08T23:13:39Z",
              "enrollments": Array [
                Object {
                  "enrollment_state": "active",
                  "role": "TeacherEnrollment",
                  "role_id": "1",
                  "type": "teacher",
                  "user_id": "1",
                },
              ],
              "id": "2",
              "image_download_url": "https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg",
              "is_favorite": false,
              "name": "Learn React Native",
              "permissions": Object {
                "viewAllGrades": true,
                "viewAnalytics": true,
              },
              "short_name": "rn",
              "term": Object {
                "created_at": "2018-02-08T23:13:39Z",
                "end_at": "2030-02-08T23:13:39Z",
                "id": "1",
                "name": "Default Term",
                "start_at": "2018-02-08T23:13:39Z",
                "workflow_state": "available",
              },
              "workflow_state": "available",
            },
            Object {
              "account_id": "1",
              "course_code": "rn 101",
              "default_view": "wiki",
              "end_at": "2030-02-08T23:13:39Z",
              "enrollments": Array [
                Object {
                  "enrollment_state": "active",
                  "role": "TeacherEnrollment",
                  "role_id": "1",
                  "type": "teacher",
                  "user_id": "1",
                },
              ],
              "id": "3",
              "image_download_url": "https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg",
              "is_favorite": false,
              "name": "Learn React Native",
              "permissions": Object {
                "viewAllGrades": true,
                "viewAnalytics": true,
              },
              "short_name": "rn",
              "term": Object {
                "created_at": "2018-02-08T23:13:39Z",
                "end_at": "2030-02-08T23:13:39Z",
                "id": "1",
                "name": "Default Term",
                "start_at": "2018-02-08T23:13:39Z",
                "workflow_state": "available",
              },
              "workflow_state": "available",
            },
          ],
          "key": 1,
          "title": "courses",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    onViewableItemsChanged={undefined}
    renderItem={[Function]}
    renderSectionHeader={[Function]}
    scrollEventThrottle={50}
    sections={
      Array [
        Object {
          "data": Array [
            Object {
              "account_id": "1",
              "course_code": "rn 101",
              "default_view": "wiki",
              "end_at": "2030-02-08T23:13:39Z",
              "enrollments": Array [
                Object {
                  "enrollment_state": "active",
                  "role": "TeacherEnrollment",
                  "role_id": "1",
                  "type": "teacher",
                  "user_id": "1",
                },
              ],
              "id": "1",
              "image_download_url": "https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg",
              "is_favorite": true,
              "name": "Learn React Native",
              "permissions": Object {
                "viewAllGrades": true,
                "viewAnalytics": true,
              },
              "short_name": "rn",
              "term": Object {
                "created_at": "2018-02-08T23:13:39Z",
                "end_at": "2030-02-08T23:13:39Z",
                "id": "1",
                "name": "Default Term",
                "start_at": "2018-02-08T23:13:39Z",
                "workflow_state": "available",
              },
              "workflow_state": "available",
            },
          ],
          "key": 0,
          "title": "favorites",
        },
        Object {
          "data": Array [
            Object {
              "account_id": "1",
              "course_code": "rn 101",
              "default_view": "wiki",
              "end_at": "2030-02-08T23:13:39Z",
              "enrollments": Array [
                Object {
                  "enrollment_state": "active",
                  "role": "TeacherEnrollment",
                  "role_id": "1",
                  "type": "teacher",
                  "user_id": "1",
                },
              ],
              "id": "2",
              "image_download_url": "https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg",
              "is_favorite": false,
              "name": "Learn React Native",
              "permissions": Object {
                "viewAllGrades": true,
                "viewAnalytics": true,
              },
              "short_name": "rn",
              "term": Object {
                "created_at": "2018-02-08T23:13:39Z",
                "end_at": "2030-02-08T23:13:39Z",
                "id": "1",
                "name": "Default Term",
                "start_at": "2018-02-08T23:13:39Z",
                "workflow_state": "available",
              },
              "workflow_state": "available",
            },
            Object {
              "account_id": "1",
              "course_code": "rn 101",
              "default_view": "wiki",
              "end_at": "2030-02-08T23:13:39Z",
              "enrollments": Array [
                Object {
                  "enrollment_state": "active",
                  "role": "TeacherEnrollment",
                  "role_id": "1",
                  "type": "teacher",
                  "user_id": "1",
                },
              ],
              "id": "3",
              "image_download_url": "https://farm3.staticflickr.com/2926/14690771011_945f91045a.jpg",
              "is_favorite": false,
              "name": "Learn React Native",
              "permissions": Object {
                "viewAllGrades": true,
                "viewAnalytics": true,
              },
              "short_name": "rn",
              "term": Object {
                "created_at": "2018-02-08T23:13:39Z",
                "end_at": "2030-02-08T23:13:39Z",
                "id": "1",
                "name": "Default Term",
                "start_at": "2018-02-08T23:13:39Z",
                "workflow_state": "available",
              },
              "workflow_state": "available",
            },
          ],
          "key": 1,
          "title": "courses",
        },
      ]
    }
    stickyHeaderIndices={
      Array [
        0,
        3,
      ]
    }
    stickySectionHeadersEnabled={true}
    updateCellsBatchingPeriod={50}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityTraits="header"
          style={
            Array [
              Object {
                "backgroundColor": "#F5F5F5",
                "flex": 1,
                "height": "auto",
                "justifyContent": "center",
                "paddingLeft": 16,
                "paddingRight": 8,
                "paddingVertical": 4,
              },
              Object {
                "borderBottomColor": "#C7CDD1",
                "borderBottomWidth": 0.5,
              },
            ]
          }
        >
          <Text
            accessibilityTraits="header"
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontFamily": ".SFUIDisplay-semibold",
                  "fontSize": 20,
                },
                Object {
                  "backgroundColor": "#F5F5F5",
                  "color": "#73818C",
                  "fontSize": 14,
                  "fontWeight": "600",
                },
              ]
            }
          >
            favorites
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <TouchableHighlight
          accessibilityLabel={undefined}
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          onPress={[Function]}
          style={
            Array [
              Object {
                "borderTopColor": "#C7CDD1",
                "borderTopWidth": 0.5,
              },
              Object {
                "borderBottomColor": "#C7CDD1",
                "borderBottomWidth": 0.5,
              },
            ]
          }
          testID="inbox.course-select.course-1"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "flex": 1,
                  "flexDirection": "row",
                  "minHeight": 54,
                  "paddingBottom": 12,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 12,
                },
                Object {
                  "backgroundColor": "white",
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                  },
                  Object {
                    "marginLeft": 0,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={0}
                style={
                  Array [
                    Object {
                      "fontFamily": ".SFUIDisplay",
                    },
                    Object {
                      "color": "#2D3B45",
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "fontWeight": "600",
                      },
                    ],
                    Object {
                      "fontFamily": ".SFUIDisplay-semibold",
                    },
                  ]
                }
              >
                Learn React Native
              </Text>
            </View>
          </View>
        </TouchableHighlight>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityTraits="header"
          style={
            Array [
              Object {
                "backgroundColor": "#F5F5F5",
                "flex": 1,
                "height": "auto",
                "justifyContent": "center",
                "paddingLeft": 16,
                "paddingRight": 8,
                "paddingVertical": 4,
              },
              Object {
                "borderBottomColor": "#C7CDD1",
                "borderBottomWidth": 0.5,
              },
            ]
          }
        >
          <Text
            accessibilityTraits="header"
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": ".SFUIDisplay",
                },
                Object {
                  "color": "#2D3B45",
                  "fontFamily": ".SFUIDisplay-semibold",
                  "fontSize": 20,
                },
                Object {
                  "backgroundColor": "#F5F5F5",
                  "color": "#73818C",
                  "fontSize": 14,
                  "fontWeight": "600",
                },
              ]
            }
          >
            courses
          </Text>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <TouchableHighlight
          accessibilityLabel={undefined}
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          onPress={[Function]}
          style={
            Array [
              Object {
                "borderTopColor": "#C7CDD1",
                "borderTopWidth": 0.5,
              },
              Object {
                "borderBottomColor": "#C7CDD1",
                "borderBottomWidth": 0.5,
              },
            ]
          }
          testID="inbox.course-select.course-2"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "flex": 1,
                  "flexDirection": "row",
                  "minHeight": 54,
                  "paddingBottom": 12,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 12,
                },
                Object {
                  "backgroundColor": "white",
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                  },
                  Object {
                    "marginLeft": 0,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={0}
                style={
                  Array [
                    Object {
                      "fontFamily": ".SFUIDisplay",
                    },
                    Object {
                      "color": "#2D3B45",
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "fontWeight": "600",
                      },
                    ],
                    Object {
                      "fontFamily": ".SFUIDisplay-semibold",
                    },
                  ]
                }
              >
                Learn React Native
              </Text>
            </View>
          </View>
        </TouchableHighlight>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <TouchableHighlight
          accessibilityLabel={undefined}
          accessibilityTraits={
            Array [
              "button",
            ]
          }
          onPress={[Function]}
          style={
            Array [
              undefined,
              Object {
                "borderBottomColor": "#C7CDD1",
                "borderBottomWidth": 0.5,
              },
            ]
          }
          testID="inbox.course-select.course-3"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "flex": 1,
                  "flexDirection": "row",
                  "minHeight": 54,
                  "paddingBottom": 12,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 12,
                },
                Object {
                  "backgroundColor": "white",
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                    "justifyContent": "center",
                  },
                  Object {
                    "marginLeft": 0,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                numberOfLines={0}
                style={
                  Array [
                    Object {
                      "fontFamily": ".SFUIDisplay",
                    },
                    Object {
                      "color": "#2D3B45",
                      "fontSize": 16,
                    },
                    Array [
                      Object {
                        "fontWeight": "600",
                      },
                    ],
                    Object {
                      "fontFamily": ".SFUIDisplay-semibold",
                    },
                  ]
                }
              >
                Learn React Native
              </Text>
            </View>
          </View>
        </TouchableHighlight>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      />
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
